sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -106,7 +106,14 @@ private[sql] class SessionState(
   // when connecting to ThriftServer.
   lazy val streamingQueryManager: StreamingQueryManager = streamingQueryManagerBuilder()
 
-  lazy val artifactManager: ArtifactManager = artifactManagerBuilder()
+  private[sql] var _artifactManager: Option[ArtifactManager] = None
```

Comment:
Why not use `artifactManagerBuilder` here?

Reply:
I want to limit the visibility of the setter.

Reply:
I am not sure I follow. Now you have mutable state that makes reasoning about the resulting artifact manager harder than it should be.

You can make the `artifactManagerBuilder ` return a constant right?

Reply:
I removed the logic of re-using ArtifactManager. This line of code won't be changed.